msal/src/main/java/com/microsoft/identity/client/controllers/MSALTokenCommand.java

```diff
@@ -0,0 +1,63 @@
+package com.microsoft.identity.client.controllers;
+
+import android.content.Context;
+import android.content.Intent;
+
+import com.microsoft.identity.client.AuthenticationCallback;
+import com.microsoft.identity.client.AuthenticationResult;
+
+public class MSALTokenCommand implements MSALTokenOperation {
+
+    private MSALOperationParameters parameters;
+    private MSALController controller;
+    private Context context;
+    private AuthenticationCallback callback;
+
+
+    @Override
+    public AuthenticationResult execute() {
+        if(getParameters() instanceof MSALAcquireTokenSilentOperationParameters){
```

Comment:
Could this type-enforcement move to the setter/getter/constructor level?

Reply:
Sure.  Fixed.